fluentd/lib/fluentd-okd-config-generator/bin/fluentd-okd-config-generator

```diff
@@ -0,0 +1,136 @@
+#!/usr/bin/env ruby
+
+require 'optparse'
+# require 'fluentd-okd-config-generator'
+require File.join(File.dirname(__FILE__), '..', 'lib/fluentd-okd-config-generator') 
+include FluentdOKDConfigGenerator::PipelineInputLoader
+
+usage = %{
+    Usage: #{File.basename(__FILE__)} -i INPUTFILE -t TARGETFILE
+    
+    This utility generates fluentd configuration for the OKD
+    pipeline feature.  It is capable of creating configurations to
+    target the following endpoint types:
+
+      * Elasticsearch
+    
+    The generator produces fluent configurations base on an YAML input file like:
+
+        ---
+        logs.app:
+          targets:
+          - type: 'elasticsearch'
+            endpoint: 'es.svc.messaging.cluster.local:9654'
+            tls_cert: '/var/run/ocp-collector/secrets/my-es-secret/cert'
+            tls_cacert: '/var/run/ocp-collector/secrets/my-es-secret/cacert'
+            tls_key: '/var/run/ocp-collector/secrets/my-es-secret/key'
+
+    The output conf is something like:
+
+       <label @OUTPUT>
+         <match **>
+           @type relabel
+           @label @LOGS_APP
+         </match>
+       </label>
+       <label @LOGS_APP>
+         <match **>
+          ...Elasticsearch output config...
+         </match>
+       </label>
+
+    Note: The '-o' option will write the '@OUTPUT' match stanzas to a file to be included statically
+          by a configuration without the label stanza.  The output will otherwise be added to TARGETFILE
+          with the label stanza
+
+    Note: 'logs.app' is the 'catch all' for messages.  It will be last in the @OUTPUT matchers
+
+}
+options = {
+    target_file: FluentdOKDConfigGenerator::InputValidator::STDOUT,
+    output_file: FluentdOKDConfigGenerator::InputValidator::STDOUT,
+    tags: {
+      'logs.app' => '**',
+      'logs.infra' => 'journal.** system.var.log** audit.log** default openshift **_openshift-*_** **_kube-*_**'
+    }
+}
+parser = OptionParser.new do |opts|
```

Comment:
formatting is inconsistent  - some places use 2 spaces for indentation, some use 4